pytest_fixtures/user_fixtures.py

```diff
@@ -0,0 +1,27 @@
+import pytest
+from fauxfactory import gen_string
+from nailgun import entities
+
+from robottelo.constants import DEFAULT_LOC
+
+
+@pytest.fixture(scope='module')
+def module_viewer_user(module_org):
+    """Custom user with viewer role for tests validating visibility of entities or fields created
+    by some other user. Created only when accessed, unlike `module_user`.
+    """
+    viewer_role = entities.Role().search(query={'search': 'name="Viewer"'})[0]
```

Comment:
Create parameterized fixture to create any role fixture and then access that fixture in this fixture.

Reply:
Even this fixture can be parametrized for role type. So that we can create module_<specific_role>_user.

e.g OrgAdmin, Admin, Viewer etc roles.

Reply:
@jyejare I don't understand this comment. What I need to do? can you help me with this one.

Reply:
@jyejare probably I will raise a separate pr for this later. For now, in my last commit, I have just renamed module_viewer_user to default_viewer_role as you suggested.

Reply:
@jameerpathan111 Cool!